src/storage/mutate/AddEdgesProcessor.cpp

```diff
@@ -100,22 +110,162 @@ void AddEdgesProcessor::process(const cpp2::AddEdgesRequest& req) {
+                                } else if (env_->checkIndexLocked(indexState)) {
+                                    LOG(ERROR) << "The index has been locked: "
+                                               << index->get_index_name();
+                                    pushResultCode(cpp2::ErrorCode::E_DATA_CONFLICT_ERROR, partId);
+                                    onFinished();
+                                    return;
+                                } else {
+                                    batchHolder->remove(std::move(oi));
+                                }
+                            }
+                        }
+                        /*
+                        * step 2 , Insert new edge index
+                        */
+                        if (nReader != nullptr) {
+                            auto ni = indexKey(partId, nReader.get(), key, index);
+                            if (!ni.empty()) {
+                                // Check the index is building for the specified partition or not.
+                                auto indexState = env_->getIndexState(spaceId_, partId);
+                                if (env_->checkRebuilding(indexState)) {
+                                    auto opKey = OperationKeyUtils::modifyOperationKey(
+                                        partId, std::move(ni));
+                                    batchHolder->put(std::move(opKey), "");
+                                } else if (env_->checkIndexLocked(indexState)) {
+                                    LOG(ERROR) << "The index has been locked: "
+                                               << index->get_index_name();
+                                    pushResultCode(cpp2::ErrorCode::E_DATA_CONFLICT_ERROR, partId);
+                                    onFinished();
+                                    return;
+                                } else {
+                                    batchHolder->put(std::move(ni), "");
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            batchHolder->put(std::move(key), std::move(retEnc.value()));
+            dummyLock.emplace_back(std::make_tuple(spaceId_,
+                                                   partId,
+                                                   edgeKey.src.getStr(),
+                                                   edgeKey.edge_type,
+                                                   edgeKey.ranking,
+                                                   edgeKey.dst.getStr()));
+        }
+        auto batch = encodeBatchValue(std::move(batchHolder)->getBatch());
+        DCHECK(!batch.empty());
+        nebula::MemoryLockGuard<EMLI> lg(env_->edgesML_.get(), std::move(dummyLock), true);
+        if (!lg) {
+            auto conflict = lg.conflictKey();
+            LOG(ERROR) << "edge conflict "
```

Comment:
Do we need to lower this log level? WARNING or VLOG?

Reply:
> Do we need to lower this log level? WARNING or VLOG?

keep it, let me change them after retry supported.

Reply:
👌